### DISCUSSION TITLE
Tailwind 3.0: any way to obtain the unpurged CSS?

### DISCUSSION THREAD

**Discussion by keanerickson-mc** [tailwindlabs/tailwindcss#6799] [Help] [answered]
I have an application that depends on having both unpurged (i.e. the full CSS file) and purged stylesheets. With 3.0 it sounds like the purge is no longer optional. How do I get a "kitchen sink" stylesheet file that contains the sum of what tailwind can offer with base + components + utilities?

Thanks

**Comment by adamwathan** [comment, marked as answer]
Hey! No way to do it I’m afraid, see this similar thread:

https://github.com/tailwindlabs/tailwindcss/discussions/6256

As experiment we tried to do it before Christmas just to see how big the CSS file would be, but the process crashed when the CSS file reached 4gb. With all of the features in 3.0 there’s just no way to generate a “full” build. Best bet is to maintain your own very large safelist, that’s essentially how we would implement it if we were to support it.

**Reply by keanerickson-mc** [reply]
Using a safelist with an empty contents array works great. Thank you for your fast response Adam!
